### PR TITLE
Add Issue and PR templates

### LINKED ISSUE #1214: Difficult to decipher GitHub issues and PRs
Problems:

* Issue assignee oftentimes needs to work to grok a given task in terms of an undocumented problem.  
* Code reviewer oftentimes needs to work to grok a given solution in terms of an undocumented problem.

Proposed solution:

Add .github ISSUE_TEMPLATE.md and PULL_REQUEST_TEMPLATE.md to offer guidance on how we can better communicate what we did or what we should do. The templates should encourage us to write issues as problem descriptions, not task descriptions, and write PR descriptions as a proposed solution to the problem described in an issue. 

If you know any existing projects that have especially good templates, can you post them here?

### PR DESCRIPTION
**Pull request by garious**
#### Problem

* Issue assignee oftentimes needs to work to grok a given task in terms of an undocumented problem.
* Code reviewer oftentimes needs to work to grok a given solution in terms of an undocumented problem.

#### Summary of Changes

Add templates that remind contributors to start every issue and PR with the context of the problem being addressed.

Fixes #1214

### CODE REVIEW

**Comment by mvines** [comment]
I'm not really a fan of this format, feels a little pointy haired.  Problem should just be the issue/PR title for example, not all issues will have a proposed solution.

I don't disagree that we can improve our efforts wrt. filling out PR/issue details.  Not sure templates will really help herd our cats though.

**Comment by garious** [comment]
A template is a guideline not a rule

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+## Problem
```
Comment (mvines) [inline .github/ISSUE_TEMPLATE.md]: four hashes instead of two would be nicer, so the headings aren’t so large

**Comment by garious** [comment]
Let's run this experiment. I'd hope the guidelines would be respected for PRs. If they aren't for issues, that'd be less surprising, but I'd hope it would be for good reason and we can use that as feedback for tweaking the template.

**Pull request merged**